### PR TITLE
MSM: add rangecheck4 and rangecheck15 lookups in constraints

### PR DESCRIPTION
**Pull request by dannywillems**
Cherry-pick from https://github.com/o1-labs/proof-systems/pull/1966

### CODE REVIEW

File: msm/src/serialization/constraints.rs [hunk 1]
```diff
@@ -15,13 +16,17 @@ pub struct Env<Fp> {
     /// folding for instance.
     pub constraints: Vec<(usize, Expr<ConstantExpr<Fp>, Column>)>,
     pub constrain_index: usize,
+    pub rangecheck4_lookups: Vec<Lookup<E<Fp>>>,
```
Comment (querolita) [inline msm/src/serialization/constraints.rs:19]: I wonder if it is really a good idea to have the lookups split into different variable names. Looks more general to have just a single lookups vector, and then all of them get pushed into it.

File: msm/src/serialization/constraints.rs [hunk 2]
```diff
@@ -63,12 +68,22 @@ impl<F: PrimeField> InterpreterEnv<F> for Env<F> {
         Column::X(3 + j)
     }
 
-    fn range_check15(&mut self, _value: &Self::Variable) {
-        // TODO
+    fn range_check15(&mut self, value: &Self::Variable) {
+        let one = ConstantExpr::from(ConstantTerm::Literal(F::one()));
+        self.rangecheck15_lookups.push(Lookup {
+            table_id: LookupTable::RangeCheck15,
+            numerator: Expr::Atom(ExprInner::Constant(one)),
```
Comment (querolita) [inline msm/src/serialization/constraints.rs:75]: I must assume you only have "static" range checks? Like, they are always added, regardless of the values of other conditionals?

File: msm/src/serialization/constraints.rs (hunk 1 above)
Comment (querolita) [reply msm/src/serialization/constraints.rs:19]: Yeah, probably we should think of a way to generalize them to just have one. And then if you really need the distinction of different tables, think of a different way to split them but "inside" the `lookup` type/vector/struct/whatever. Maybe for a follow up.

**Review by querolita — APPROVED** [review]

File: msm/src/serialization/constraints.rs (hunk 2 above)
Comment (dannywillems) [reply msm/src/serialization/constraints.rs:75]: Yes

File: msm/src/serialization/constraints.rs (hunk 1 above)
Comment (dannywillems) [reply msm/src/serialization/constraints.rs:19]: Yes, agreed. This is ugly atm, and we must generalize. Let's do it quickly after everything works.

**Pull request merged**